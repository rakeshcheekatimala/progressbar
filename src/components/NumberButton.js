import React from 'react';
import PropTypes from 'prop-types';
import { displayNumberSign } from '../util/format';

const NumberButton = (props) => {
	let { value, onClick } = props;
	return (
		<button className="NumberButton" data-testid="numberbutton" value={value} onClick={onClick}>{displayNumberSign(value)}</button>
	)
}


NumberButton.propTypes = {
	value: PropTypes.number.isRequired,
	onClick: PropTypes.func.isRequired
}

export default NumberButton;