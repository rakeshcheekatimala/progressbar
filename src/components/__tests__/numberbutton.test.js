import React from "react";
import ReactDOM from "react-dom";
import NumberButton from "./../NumberButton";

import { render, cleanup } from "@testing-library/react";

import renderer from "react-test-renderer";

afterEach(cleanup);

it("renders without crashing", () => {
	const div = document.createElement("div");
	ReactDOM.render(<NumberButton value={5} onClick={() => { }} />, div);
});

it("renders button correctly", () => {
	const { getByTestId } = render(<NumberButton value={5} onClick={() => { }} />);
	expect(getByTestId("numberbutton")).toHaveTextContent("+5");
});

it("renders button correctly", () => {
	const { getByTestId } = render(<NumberButton value={-5} onClick={() => { }} />);
	expect(getByTestId("numberbutton")).toHaveTextContent("-5");
});

it("matches snapshot 1", () => {
	const tree = renderer.create(<NumberButton value={-5} onClick={() => { }} />).toJSON();
	expect(tree).toMatchSnapshot();
});


