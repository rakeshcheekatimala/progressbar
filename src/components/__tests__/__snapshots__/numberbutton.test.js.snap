// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot 1 1`] = `
<button
  className="NumberButton"
  data-testid="numberbutton"
  onClick={[Function]}
  value={-5}
>
  -5
</button>
`;
